### PR TITLE
New theme: Shiro's Theme

### PR DESCRIPTION
**Pull request by NightNutSky**
Well, um... New theme. Since I created it, I named it "Shiro's Theme" (obviously). I hope this will be approved. In other cases, if the name does not suit you, we can come up with it together.
Other small changes are subfolder-style sorting in the sidebar for the Guides and Premium categories.
image
image
image
image
image
image
image
image

### CODE REVIEW

File: src/theme/book.js [hunk 1]
```diff
@@ -296,13 +296,13 @@ function playpen_text(playpen) {
...
 
-        if (theme == 'coal' || theme == 'navy' || theme == 'bdfd') {
+        if (theme == 'coal' || theme == 'navy' || theme == 'bdfd' || theme == "shiro") {
             stylesheets.ayuHighlight.disabled = true;
             stylesheets.tomorrowNight.disabled = false;
             stylesheets.highlight.disabled = true;
 
             ace_theme = "ace/theme/tomorrow_night";
-        } else if (theme == 'ayu') {
+        } else if (theme == 'ayu' || theme == 'shiro') {
```
Comment (MineBartekSA) [inline src/theme/book.js:305]: This is useless and will never run.

**Review by MineBartekSA — CHANGES_REQUESTED** [review]
Thanks for the new theme.
Please remove the useless condition.

File: src/theme/book.js (hunk 1 above)
Comment (noituri) [reply src/theme/book.js:305]: this

**Review by noituri — CHANGES_REQUESTED** [review]

**Comment by NightNutSky** [comment]
Due to my poor knowledge of the JS, I don't quite understand why it won't.
But, it turns out, it should be changed like this?
```js
if (...) {
...} else if (theme = "ayu") {
...} else if (theme = "shiro") {
...} else {
...}
```
Although, when I tested locally, I did not seem to see any problems.

**Comment by NightNutSky** [comment]
Nvm

**Comment by MineBartekSA** [comment]
The problem is that you already added `theme == "shiro"` to the condition of the base if statement after an || aka or.
If you do something like:
```js
a = "b";
if (a == "b") {
    // This will run
} else if (a == "b") { // This condition will be only checked if the preceding condition is false
    // This will never run
}
```
Only the first condition that returns true will run its block.

This shouldn't really matter, it's just useless code, plus I pointed this out to make sure you are doing what you are intending to do in that if.

**Review by noituri — APPROVED** [review]

**Review by MineBartekSA — APPROVED** [review]

**Pull request merged**